backend/src/Service/Notification/NotificationFirebaseService.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+
+namespace App\Service\Notification;
+
+use App\Manager\Notification\NotificationFirebaseManager;
+use App\Service\Notification\NotificationTokensService;
+use App\Service\User\UserService;
+use Kreait\Firebase\Contract\Messaging;
+use Kreait\Firebase\Messaging\CloudMessage;
+use Kreait\Firebase\Messaging\Notification;
+use App\Constant\Notification\NotificationFirebaseConstant;
+use App\Constant\Notification\NotificationConstant;
+use App\Constant\Order\OrderStateConstant;
+use App\Request\Notification\NotificationFirebaseByUserIdRequest;
+use App\Request\Notification\NotificationFirebaseFromAdminRequest;
+use App\Constant\Notification\NotificationTokenConstant;
+
+class NotificationFirebaseService
+{
+    private Messaging $messaging;
+    private NotificationFirebaseManager $notificationFirebaseManager;
+    private NotificationTokensService $notificationTokensService;
+    private UserService $userService;
+
+    public function __construct(Messaging $messaging, NotificationFirebaseManager $notificationFirebaseManager, NotificationTokensService $notificationTokensService, UserService $userService)
+    {
+        $this->messaging = $messaging;
+        $this->notificationFirebaseManager = $notificationFirebaseManager;
+        $this->notificationTokensService = $notificationTokensService;
+        $this->userService = $userService;
+    }
+
+    public function notificationToCaptains(int $orderId)
+    {
+        $getTokens = $this->notificationTokensService->getUsersTokensByAppType(NotificationTokenConstant::APP_TYPE_CAPTAIN);
+
+        $tokens = [];
+
+        foreach ($getTokens as $token) {
+            $tokens[] = $token['token'];
+        }
+
+        $payload = [
+            'click_action' => 'FLUTTER_NOTIFICATION_CLICK',
+            'navigate_route' => NotificationFirebaseConstant::URL,
+            'argument' => $orderId,
+        ];
+
+        $message = CloudMessage::new()
+            ->withNotification(
+                Notification::create(NotificationFirebaseConstant::DELIVERY_COMPANY_NAME, NotificationFirebaseConstant::MESSAGE_CAPTAIN_NEW_ORDER))
+            ->withDefaultSounds()
+            ->withHighestPossiblePriority()->withData($payload);
+        $this->messaging->sendMulticast($message, $tokens);
```

Comment:
I'm just wondering, if the function's aim is sending the notification to a single user, why we are using multicast option?